core/app/modules/site-editor/assets/js/templates/layout.js

```diff
@@ -7,12 +7,15 @@ import './site-editor.scss';
 export default function Layout( props ) {
 	const config = {
 		title: __( 'Theme Builder', 'elementor' ),
-		titleRedirectRoute: '/site-editor',
```

Comment:
I would consider using
`titleRedirectRoute: props.titleRedirectRoute ?? null,`
instead of the `if`, if the `null` works as intended here.